### PR TITLE
[Confluence] Add E2E test for the connector

### PR DESCRIPTION
**Pull request by praveen-elastic**
#### Pre-Review Checklist
- [x] ~~Covered the changes with automated tests~~
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [x] ~~Considered corresponding documentation changes~~
- [x] Contributed any configuration settings changes to the configuration reference: Added Confluence to `nightly.yml`

#### Changes Requiring Extra Attention

- [x] New external dependency added for flask

### CODE REVIEW

File: connectors/sources/tests/fixtures/confluence/fixture.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    args = request.args
+    content = {
+        "results": [],
+        "start": 0,
+        "limit": 100,
+        "size": 100,
+        "_links": {"next": None},
+    }
+    document_type = args.get("type", "page")
+    for content_count in range(100):
```
Comment (timgrein) [inline connectors/sources/tests/fixtures/confluence/fixture.py:91]: I would suggest, that we use a variable with a useful name for `100`.

File: connectors/sources/tests/fixtures/confluence/fixture.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    """
+    attachments = {
+        "results": [],
+        "start": 0,
+        "limit": 100,
+        "size": 5,
+        "_links": {"next": None},
+    }
+
+    for attachment_count in range(1, 6):
```
Comment (timgrein) [inline connectors/sources/tests/fixtures/confluence/fixture.py:127]: I would also suggest here to use variables for `1` and `6`.

**Review by timgrein — COMMENTED** [review]
I'm not 100% sure, if that's the whole test right? That's more like setting up a fake API?

File: connectors/sources/tests/fixtures/confluence/fixture.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        spaces["results"].append(
+            {
+                "id": "space 0",
+                "name": "Demo Space 0",
+                "_links": {
+                    "webui": "/spaces/space0",
+                },
+            }
+        )
+    else:
```
[the remark this answers is not in the document]
Comment (praveen-elastic) [reply connectors/sources/tests/fixtures/confluence/fixture.py]: We are using this for ping api call, by keeping limit as 1.

**Comment by timgrein** [comment]
@praveen-elastic Makes sense. But still this PR does not include the actual test right or am I missing something?

**Comment by praveen-elastic** [comment]
> @praveen-elastic Makes sense. But still this PR does not include the actual test right or am I missing something?

This is the complete set of changes for E2E to run, we haven't added `load`, `remove` methods but instead mocking the APIs and passing the response. Morover, we still generate data for attachments via `LARGE_DATA = "".join([random.choice(string.ascii_letters) for _ in range(FILE_SIZE)])`

**Review by timgrein — APPROVED** [review]
LGTM

**Pull request merged**